tests/input.spec.ts

```diff
@@ -0,0 +1,210 @@
+  await expect(inputEmail.locator('input').first()).toHaveAttribute('type', 'email')
+});
+
+test('Renders without label if not given a label', async ({ page }) => {
+  const inputText = await page.getByTestId('input-text')
+  await expect(await inputText.locator(':above(input)').first().textContent()).toBe(" ")
+});
+
+test('Given attribute tooltip with no state, render info icon above the input', async ({ page }) => {
+  const inputTooltip = page.getByTestId("input-tooltip-default")
+  await expect(inputTooltip.locator('v-tooltip > div').first()).toHaveClass(/icon-info-outline/)
+});
+
+test('Given attribute tooltip with info state, render info icon above the input', async ({ page }) => {
+  const inputInfoTooltip = page.getByTestId("input-tooltip-info")
+  await expect(inputInfoTooltip.locator('v-tooltip > div').first()).toHaveClass(/icon-info-outline/)
+});
+
+test('Given attribute tooltip with warn state, render warn icon above the input', async ({ page }) => {
+  const inputWarnTooltip = page.getByTestId("input-tooltip-warn")
+  await expect(inputWarnTooltip.locator('v-tooltip > div').first()).toHaveClass(/icon-error-outline text-warning-fg/)
+});
+
+test('Given attribute tooltip with error state, render error icon above the input', async ({ page }) => {
+  const inputErrorTooltip = page.getByTestId("input-tooltip-error")
+  await expect(inputErrorTooltip.locator('v-tooltip > div').first()).toHaveClass(/icon-error-outline text-danger-fg/)
+});
+
+test('Given a readonly attribute, renders readonly input', async ({ page }) => {
+  const inputReadonly = page.getByTestId("input-readonly")
+  await expect(inputReadonly).toBeVisible()
+  await expect(inputReadonly.locator('input').first()).toHaveAttribute('readonly', '')
+  await expect(inputReadonly.locator('input').first()).toHaveValue('teehee')
+});
+
+test('Given a disabled attribute, renders disabled input', async ({ page }) => {
+  const inputDisabled = page.getByTestId("input-disabled")
+  await expect(inputDisabled).toBeVisible()
+  await expect(inputDisabled.locator('input').first()).toBeDisabled()
+  await expect(inputDisabled.locator('input').first()).toHaveClass(/bg-disabled-bg text-disabled-fg border-disabled-bg/)
+});
+
+test('Displays message below input box', async ({ page }) => {
+  const inputMessage = page.getByTestId("input-message")
+  await expect(inputMessage).toBeVisible()
+  await expect(inputMessage.getByText('it\'s the cliiimb')).toBeVisible()
+  await expect(inputMessage.locator('input:above(:text("it\'s the cliiimb"))')).toBeVisible()
+});
+
+test('With error state, displays message below input box in red', async ({ page }) => {
+  const inputMessageError = page.getByTestId("input-message-error")
+  await expect(inputMessageError.getByText('fell off the cliiimb')).toHaveClass(/text-red-600/)
+  await expect(inputMessageError.locator('input:above(:text("fell off the cliiimb"))')).toBeVisible()
+});
+
+test('Fires input event with value on input', async ({ page }) => {
```

Comment:
for type=number, new input cleaning does not fire an event for every keystroke but rather only on valid formatted numbers, will that be a problem?

Reply:
that sounds fine